src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2899,6 +2900,60 @@ private void API_AuxiliaryAspects()
             return (cc, tt) -> time;
         });
 
+        this.expr.addLazyFunction("unix_time",0,(c,t,lv)->{
+            Value time = new NumericValue(System.currentTimeMillis());
+            return (_c,_t)-> time;
+        });
+
+        this.expr.addLazyFunction("convert_date",-1,(c,t,lv)->{
+            for (int i = 0; i<lv.size();i++){
+                if(!(lv.get(i).evalValue(c) instanceof NumericValue))
```

Comment:
you can't evaluate multiple times the same expression. Once you call .evalValue on an expression, you can't touch it again, unless you are implementing a loop. This is why most functions check arguments one by one, and act accordingly. convert_date should evaluate first argument, and then if its a number - perform to-date conversion, if its a list - get all its elements and perform from-date conversion. optionally if there is more than one argument - perform from-date conversion based on the arguments alone.

Reply:
what is that in english?

Reply:
ok, I think I got it, but only 1 argument, list or numeric.